frameworks/CSharp/aspnetcore/Benchmarks/Middleware/MultipleUpdatesRavenMiddleware.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Benchmarks.Configuration;
+using Benchmarks.Data;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace Benchmarks.Middleware
+{
+    public class MultipleUpdatesRavenMiddleware
+    {
+        private static readonly PathString _path = new PathString(Scenarios.GetPath(s => s.DbMultiUpdateRaven));
+        private static readonly JsonSerializerSettings _jsonSettings = new JsonSerializerSettings
+        {
+            ContractResolver = new CamelCasePropertyNamesContractResolver()
+        };
+
+        private readonly RequestDelegate _next;
+        private readonly RavenDb _db;
+
+        public MultipleUpdatesRavenMiddleware(RequestDelegate next, RavenDb provider)
+        {
+            _next = next;
+            _db = provider;
+        }
+
+        public async Task Invoke(HttpContext httpContext)
+        {
+            if (httpContext.Request.Path.StartsWithSegments(_path, StringComparison.Ordinal))
+            {
+                var count = MiddlewareHelpers.GetMultipleQueriesQueryCount(httpContext);
+
+                var rows = await _db.LoadMultipleUpdatesRows(count);
```

Comment:
Is there specific format required? We can use the streaming option to pass the raw results directly to the client, bypassing any work on the server side.

Reply:
Yes, format is specified. There are even ordering issues AFAIK for streaming. Multiple updates is actually a Bulk Insert, haven't code it like that yet, so no streaming.

Reply:
Might be better to see if we can do that directly via a query and stream it, yen.